x-pack/platform/plugins/shared/streams/server/lib/streams/client.ts

```diff
@@ -206,26 +212,64 @@ export class StreamsClient {
 
     const { dashboards, queries } = request;
 
-    const queryLinks = queries.map((query) => ({
-      [ASSET_ID]: query.id,
-      [ASSET_TYPE]: 'query' as const,
-      query,
-    }));
-
-    await this.dependencies.assetClient.syncAssetList(stream.name, [
-      ...dashboards.map((dashboard) => ({
+    // sync dashboards as before
+    await this.dependencies.assetClient.syncAssetList(
+      stream.name,
+      dashboards.map((dashboard) => ({
         [ASSET_ID]: dashboard,
         [ASSET_TYPE]: 'dashboard' as const,
       })),
-      ...queryLinks,
-    ]);
+      'dashboard'
```

Comment:
💬 syncAssetList will remove every assets not provided. And since we want to handle the query assets slightly differently I had to add a filter on what assets to sync. I'll probably change this to an array of asset type since we might have more than just dashboard and query.